src/Subsystem/SchedulerSubsystem/LoadingPassengerState.java

```diff
@@ -0,0 +1,80 @@
+package Subsystem.SchedulerSubsystem;
+
+import Messaging.Messages.Commands.MoveElevatorCommand;
+import Messaging.Messages.Events.DestinationEvent;
+import Messaging.Messages.Events.ElevatorStateEvent;
+import Messaging.Messages.Events.FloorRequestEvent;
+import Messaging.Messages.Events.PassengerLoadEvent;
+import Messaging.Messages.SystemMessage;
+import Messaging.Messages.Commands.SendPassengersCommand;
+import Messaging.Messages.Commands.MovePassengersCommand;
+import StatePatternLib.Context;
+import StatePatternLib.State;
+import com.sun.jdi.InvalidTypeException;
+
+/**
+ * Scheduler FSM State: Loading Passenger State.
+ *
+ * Class which models the Loading Passenger State of the Scheduler FSM.
+ *
+ * Responsibilities:
+ * - Command an Elevator to load Passengers.
+ *
+ * @author MD
+ * @version Iteration-3
+ */
+public class LoadingPassengerState extends State {
+
+    /* Instance Variables */
+    
+    // The PassengerLoadEvent to process
+    private PassengerLoadEvent event;
+
+    /**
+     * Parametric constructor.
+     *
+     * @param context Context of state machine that this is a state of.
+     * @author MD
+     */
+    public LoadingPassengerState(Context context, PassengerLoadEvent event) {
+        super(context);
+        // Initialize the PassengerLoadEvent that needs processing in this state.
+        this.event = event;
+    }
+
+    /**
+     * Entry activities for this state.
+     */
+    @Override
+    public void entry() {
+        System.out.println("[INFO::FSM] Scheduler:LoadingPassengerState:Entry");
+    }
+
+    /**
+     * Do activities for this state.
+     *
+     * In this state, we just transmit a MovePassengersCommand to an Elevator,
+     * and then return to Receiving state.
+     */
+    @Override
+    public void doActivity() {
+        System.out.println("[INFO::FSM] Scheduler:LoadingPassengerState:Do");
+
+        // Load Passengers, notify Elevator
+        MovePassengersCommand movePassengersCommand = new MovePassengersCommand(event.elevNumber(), event.passengers());
+        ((Scheduler)context).transmitToElevator(movePassengersCommand);
+
+        // Next State: ReceivingState
+        // Required Constructor Arguments: context
+        context.setNextState(new ReceivingState(context));
```

Comment:
similar idea of setting the next state in the exit assuming that it is indenpendant of the do activity results.

Reply:
Agreed again! Changes made.